crates/typst/src/layout/inline/line.rs

```diff
@@ -486,6 +492,19 @@ pub fn commit(
 
     // Build the frames and determine the height and baseline.
     let mut frames = vec![];
+
+    if let Some(numbering) = ParLine::numbering_in(styles) {
+        let par_line = ParLineMarker::new(numbering).pack();
+        let par_line_region =
+            Regions::one(Size::new(width, full), Axes::new(true, false));
+        frames.push((
+            offset,
+            par_line
+                .layout(engine, locator.next(&par_line), styles, par_line_region)?
+                .into_frame(),
+        ));
```

Comment:
IIUC this only generates a Frame with a single `Tag`. Could we generate the tag directly?

Going through the full layout machinery feels a bit wasteful just to generate a `Tag` whose location doesn't even really interest us. If I'm not mistaken, the location deduplication in the flow is unnecessary (as mentioned in another comment). I that case we could also generate a somewhat fake Tag.

I'm not sure here, I think the other comments should be addressed first and we can discuss this again later.

Reply:
I inlined the location generation for the tag, and it seems to be working. Let me know if the new solution is appropriate.